apps/blockchain/lib/blockchain/block.ex

```diff
@@ -255,12 +258,14 @@ defmodule Blockchain.Block do
   @spec put_block(t, TrieStorage.t(), binary() | nil) :: {:ok, {EVM.hash(), TrieStorage.t()}}
   def put_block(block, trie, predefined_key \\ nil) do
     hash = if predefined_key, do: predefined_key, else: hash(block)
+
     block_rlp = block |> serialize |> ExRLP.encode()
 
     updated_trie =
       trie
       |> TrieStorage.put_raw_key!(hash, block_rlp)
-      |> TrieStorage.put_raw_key!(get_block_hash_key(block.header.number), hash)
+      |> TrieStorage.put_raw_key!(block_hash_key(block.header.number), hash)
+      |> put_additional_info(hash, block, block_rlp)
```

Comment:
Since this is also indexed by block hash, should we just store blocks as `[block_rlp, additional_info]`? Since I imagine they'll be fetched sequentially frequently, might consider just storing them together to prevent additional calls to disk to store / lookup. 

This would also allow you to add the additional properties directly to `t()` instead of as nullable `additional_info`

Reply:
I'll address it in the next PR. We won't store RLP anymore